src/main/java/cart/service/ItemService.java

```diff
@@ -1,40 +1,43 @@
 package cart.service;
 
+import cart.controller.dto.request.ItemRequest;
+import cart.controller.dto.response.ItemResponse;
+import cart.dao.ItemDao;
 import cart.domain.ImageUrl;
+import cart.domain.Item;
 import cart.domain.Name;
 import cart.domain.Price;
-import cart.controller.dto.ItemRequest;
-import cart.controller.dto.ItemResponse;
-import cart.dao.ItemDao;
-import cart.domain.Item;
 import cart.exception.NotFoundResultException;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Service
+@Transactional
```

Comment:
Service에 Transactional 어노테이션 붙이게 되면 RuntmeException 혹은 오류가 발생한 경우 롤백을 다른 예외가 없을 경우 커밋을 수행해 준다고 알고 있습니다.

readOnly=true에 대한 부분은 공식 문서를 한번 살펴보았습니다.
공식 문서에는 아래와 같이 작성되어 있었습니다.

> This just serves as a hint for the actual transaction subsystem; it will not necessarily cause failure of write access attempts. A transaction manager which cannot interpret the read-only hint will not throw an exception when asked for a read-only transaction but rather silently ignore the hint.
> 

즉, 실제로 readOnly여서 조회 외에는 불가능하게 만드는 것은 아니만, 해당 메서드의 동작에 대한 힌트를 주는 것입니다.
그래서 클래스에 `@Transactional (readOnly=true)`를 붙여주어도 모든 메서드는 수행이 되긴 합니다.

하지만, 읽기에 최적화하므로 변경 여부에 대해 신경을 쓰지 않아 오류가 발생해도 롤백되지 않고 넘어가게 될 수도 있습니다.
대신 읽기에 있어선 트랜잭션 범위 최소화와 같인 불필요한 작업을 줄이기에 보다 성능이 향상된다는 장점이 있습니다.
또한, readOnly=true를 붙임으로써 해당 메서드의 역할의 힌트를 주어 좀 더 직관적으로 알 수 있다는 장점이 있습니다.

그런데 조사하다 보니 조회 외에는 아무 작업도 하지 수행하지 못한다는 이야기가 계속해 나오는데 제가 잘못 이해한 것인지 궁금합니다..!

Reply:
DBSM별로 트랜잭션 어노테이션이 반영되는 방식이 다를 수 있기 때문에 똑같은 어노테이션이어도 다른 결과가 나올 수 있습니다 🙂
관련해서 참고하기 [좋은 글](https://lob-dev.tistory.com/entry/DBMS-%EB%B3%84-Transaction-Read-Only%EC%97%90-%EB%8C%80%ED%95%9C-%EB%8F%99%EC%9E%91-%EB%B0%A9%EC%8B%9D-1)이 있으니 한번 읽어보면 좋을 것 같네요!